ipq807x: add support for Yuncore AX880

legale opened the pull request:
SPECIFICATION:
    - Chipset:  IPQ8072A +QCN5054+QCN5024+QCA8081*2
    - Flash NOR-8MB AND NAND-128MB
    - RAM 1Gb DDR
    - IEEE 802.11:  802.11ax/ac/b/g/n
    - 4*4 2.4G Wi-Fi standard   802.11b/g/n/ax
    - 4*4 5.8G Wi-Fi Standard   802.11 a/n/ac/ax
    - 2x 1 Gbps Ethernet (QCA8081) with 802.3at PoE input support
    - 1x  Reset
    - 1x  Bluetooth(optional)
    - 1x  DC Port 12V 3A
    - 4x Antenna    IPEX Connector, 3dBi omni antennas
    - Data Rate:    3657Mbps ( 2.4G: 1182Mbps (11ax 4x4); 5.8G: 2475Mbps (11ax 4x4))
    - RF Power: 2.4g ≤ 20dBm; 5.8g ≤ 19dBm
    - LED light:    Sys; 5.8G wifi; 2.4G wifi; WAN; LAN
    - Max Power Consumption:    ≤ 22W
    - Size: 198mm * 198mm * 41.02mm

BACKUP YOUR STOCK FIRMWARE:
```
export device=ax880
mkdir -p /tmp/fw_dump_$device
cd /tmp/fw_dump_$device
dmesg > dmesg_$device.log
dtc -I fs /sys/firmware/devicetree/base > $device.dts
cat /proc/device-tree/model > model
cat /proc/mtd > proc_mtd
while read p; do
mtd_dev=$(echo $p | cut -d: -f1)
echo $mtd_dev
dd if=/dev/$mtd_dev of=$mtd_dev
done < proc_mtd
md5sum * > md5sum.log
tar -cvzf ../$device.tar.gz .
export sum=$(md5sum /tmp/$device.tar.gz | cut -d' ' -f1)
mv ../$device.tar.gz /tmp/${device}_${sum}.tar.gz
echo fw backup saved to: /tmp/${device}_${sum}.tar.gz
```
Upload your backup via tftp to the safe place.

INSTALLATION:
1. stock firmware web ui
Rename factory.bin fw image file to factory.ubin. Flash this image
like ordinary stock fw upgrade.

2. stock firmware telnet method
Enter telnet cli (login: root, password: 476t*_f0%g09y) and upload
 factory.bin fw image and rename it to factory.ubin
`cd /tmp && wget <your_web_server_ip>/factory.ubin`
`sysupgrade factory.ubin

3. initramfs method
    Put openwrt-ipq807x-generic-yuncore_ax880-initramfs-uImage.itb to your
    TFTP server and rename it to ax880.initram
    Enable serial console and enter to the u-boot cli.
    Exec these commands:
    `tftpboot <your_tftp_server_ip>:ax880.initram`
    `dhcp`

    When downloading is finished:
    `bootm`
    After booting the device, you need to upload to the device factory.ubi fw image.
    ```
    cd /tmp && wget <your_web_server_ip>/factory.ubi`
    export rootfs=$(cat /proc/mtd | grep rootfs | cut -d: -f1)
    export rootfs_1=$(cat /proc/mtd | grep rootfs_1 | cut -d: -f1)
    ubiformat /dev/${rootfs} -y -f factory.ubi
    ubiformat /dev/${rootfs_1} -y -f factory.ubi
    reboot
    ```

4. u-boot factory.ubi image method
    Put openwrt-ipq807x-generic-yuncore_ax880-squashfs-factory.ubi to your
    TFTP server and rename it to ax880.ubi
    Enter u-boot cli and exec these commands:
    `tftpboot <your_tftp_server_ip>:ax880.ubi`
    `dhcp`
    After downloading is finished:
    `flash rootfs`
    `flash rootfs_1`
    `reset`

5. u-boot factory.bin method
    Put openwrt-ipq807x-generic-yuncore_ax880-squashfs-factory.bin to your
    TFTP server and rename it to ax880.bin
    Enter u-boot cli and exec these commands:
    `tftpboot <your_tftp_server_ip>:ax880.bin`
    `dhcp`
    After downloading is finished:
    `imgaddr=$fileaddr`
    Erase rootfs memory:
    `nand erase 0x00000000 0x03400000`
    Write rootfs:
    `nand write $fileaddr 0x00000000 $filesize`
    Erase rootfs_1 memory:
    `nand erase 0x3c00000 0x3400000`
    Write rootfs_1
    `nand write $fileaddr 0x3c00000 $filesize`
    `reset`

STOCK FIRMWARE RECOVERY:
Boot initramfs image.
Upload your rootfs mtd partition to the device using scp or download
it from the device using wget.
Enter device ssh cli and exec:
```
cd /tmp && wget <your_web_server_ip>/mtd21`
export rootfs=$(cat /proc/mtd | grep rootfs | cut -d: -f1)
export rootfs_1=$(cat /proc/mtd | grep rootfs_1 | cut -d: -f1)
ubiformat /dev/${rootfs} -y -f /tmp/mtd21
ubiformat /dev/${rootfs_1} -y -f /tmp/mtd21
reboot
```

Signed-off-by: Isaev Ruslan <legale.legale@gmail.com>

legale committed:
ipq807x: add support for Yuncore AX880

SPECIFICATION:
    - Chipset:  IPQ8072A +QCN5054+QCN5024+QCA8081*2
    - Flash NOR-8MB AND NAND-128MB
    - RAM 1Gb DDR
    - IEEE 802.11:  802.11ax/ac/b/g/n
    - 4*4 2.4G Wi-Fi standard   802.11b/g/n/ax
    - 4*4 5.8G Wi-Fi Standard   802.11 a/n/ac/ax
    - 2x 1 Gbps Ethernet (QCA8081) with 802.3at PoE input support
    - 1x  Reset
    - 1x  Bluetooth(optional)
    - 1x  DC Port 12V 3A
    - 4x Antenna    IPEX Connector, 3dBi omni antennas
    - Data Rate:    3657Mbps ( 2.4G: 1182Mbps (11ax 4x4); 5.8G: 2475Mbps (11ax 4x4))
    - RF Power: 2.4g ≤ 20dBm; 5.8g ≤ 19dBm
    - LED light:    Sys; 5.8G wifi; 2.4G wifi; WAN; LAN
    - Max Power Consumption:    ≤ 22W
    - Size: 198mm * 198mm * 41.02mm

BACKUP YOUR STOCK FIRMWARE:
```
export device=ax880
mkdir -p /tmp/fw_dump_$device
cd /tmp/fw_dump_$device
dmesg > dmesg_$device.log
dtc -I fs /sys/firmware/devicetree/base > $device.dts
cat /proc/device-tree/model > model
cat /proc/mtd > proc_mtd
while read p; do
mtd_dev=$(echo $p | cut -d: -f1)
echo $mtd_dev
dd if=/dev/$mtd_dev of=$mtd_dev
done < proc_mtd
md5sum * > md5sum.log
tar -cvzf ../$device.tar.gz .
export sum=$(md5sum /tmp/$device.tar.gz | cut -d' ' -f1)
mv ../$device.tar.gz /tmp/${device}_${sum}.tar.gz
echo fw backup saved to: /tmp/${device}_${sum}.tar.gz
```
Upload your backup via tftp to the safe place.

INSTALLATION:
1. stock firmware web ui
Rename factory.bin fw image file to factory.ubin. Flash this image
like ordinary stock fw upgrade.

2. stock firmware telnet method
Enter telnet cli (login: root, password: 476t*_f0%g09y) and upload
 factory.bin fw image and rename it to factory.ubin
`cd /tmp && wget <your_web_server_ip>/factory.ubin`
`sysupgrade factory.ubin

3. initramfs method
    Put openwrt-ipq807x-generic-yuncore_ax880-initramfs-uImage.itb to your
    TFTP server and rename it to ax880.initram
    Enable serial console and enter to the u-boot cli.
    Exec these commands:
    `tftpboot <your_tftp_server_ip>:ax880.initram`
    `dhcp`

    When downloading is finished:
    `bootm`
    After booting the device, you need to upload to the device factory.ubi fw image.
    ```
    cd /tmp && wget <your_web_server_ip>/factory.ubi`
    export rootfs=$(cat /proc/mtd | grep rootfs | cut -d: -f1)
    export rootfs_1=$(cat /proc/mtd | grep rootfs_1 | cut -d: -f1)
    ubiformat /dev/${rootfs} -y -f factory.ubi
    ubiformat /dev/${rootfs_1} -y -f factory.ubi
    reboot
    ```

4. u-boot factory.ubi image method
    Put openwrt-ipq807x-generic-yuncore_ax880-squashfs-factory.ubi to your
    TFTP server and rename it to ax880.ubi
    Enter u-boot cli and exec these commands:
    `tftpboot <your_tftp_server_ip>:ax880.ubi`
    `dhcp`
    After downloading is finished:
    `flash rootfs`
    `flash rootfs_1`
    `reset`

5. u-boot factory.bin method
    Put openwrt-ipq807x-generic-yuncore_ax880-squashfs-factory.bin to your
    TFTP server and rename it to ax880.bin
    Enter u-boot cli and exec these commands:
    `tftpboot <your_tftp_server_ip>:ax880.bin`
    `dhcp`
    After downloading is finished:
    `imgaddr=$fileaddr && nand device 0`
    Erase rootfs memory:
    `nand erase 0x00000000 0x03400000`
    Write rootfs:
    `nand write $fileaddr 0x00000000 $filesize`
    Erase rootfs_1 memory:
    `nand erase 0x3c00000 0x3400000`
    Write rootfs_1
    `nand write $fileaddr 0x3c00000 $filesize`
    `reset`

STOCK FIRMWARE RECOVERY:
Boot initramfs image.
Upload your rootfs mtd partition to the device using scp or download
it from the device using wget.
Enter device ssh cli and exec:
```
cd /tmp && wget <your_web_server_ip>/mtd21`
export rootfs=$(cat /proc/mtd | grep rootfs | cut -d: -f1)
export rootfs_1=$(cat /proc/mtd | grep rootfs_1 | cut -d: -f1)
ubiformat /dev/${rootfs} -y -f /tmp/mtd21
ubiformat /dev/${rootfs_1} -y -f /tmp/mtd21
reboot
```

Signed-off-by: Isaev Ruslan <legale.legale@gmail.com>
Reviewed-by: Robert Marko <robimarko@gmail.com>

musashino205 commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,390 @@
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+        bootargs-append = " root=/dev/ubiblock0_1";
```
Please use a tab instead of spaces for indentation.

legale replied:
Done.

robimarko replied:
Still broken indentation

legale replied:
> Still broken indentation

fixed.

musashino205 commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,390 @@
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+   
+    ethernet-phy@0 {
+        reg = <0>;
+    };
+    ethernet-phy@1 {
+        reg = <1>;
+    };
+    ethernet-phy@2 {
+        reg = <2>;
+    };
+    ethernet-phy@3 {
+        reg = <3>;
+    };
```
Please use tabs instead of spaces for indentation.

legale replied:
Done.

robimarko replied:
Not done, indentation is still broken.

robimarko replied:
Also, do these PHY-s even exist as they are not being used and you specify only 2x QCA8081 in the specs?

kirdesde replied:
As robimarko wrote, you can remove ethernet-phy@[0-3]. The device has only 2 QCA8081 phys.
port@[0-3] properties in the switch node can be removed as well.

legale replied:
> As robimarko wrote, you can remove ethernet-phy@[0-3]. The device has only 2 QCA8081 phys. port@[0-3] properties in the switch node can be removed as well.

removed.

legale replied:
> Also, do these PHY-s even exist as they are not being used and you specify only 2x QCA8081 in the specs?

I hope to use these emtpy entries somehow in qca-ssdk. There is a spam messages related to these ports at the moment. Removed now.

kirdesde replied:
You mean these log entries?

```
phy_id = 0xffffffff phytype doesn't match
```
Those can be ignored, they should only occur during ssdk init.

robimarko replied:
Yeah, those dont matter at all, its just QCA-s stupid expectation that 0-4 will always be present but its harmless.

legale replied:
I'm talking about these log entries:
![image](https://github.com/openwrt/openwrt/assets/32621719/0cd2d7b4-f9b1-4320-821b-0868f57b5565)


BTW any ideas how to fix this better than just remove print syslog in qca-ssdk?

grozzie2 replied:
> @robimarko @kirdesde I'm talking about these log entries: ![image](https://user-images.githubusercontent.com/32621719/280336330-0cd2d7b4-f9b1-4320-821b-0868f57b5565.png)
> 
> BTW any ideas how to fix this better than just remove print syslog in qca-ssdk?

I have the indoor version of this ap.

I have built with this PR included, and those spam messages are very annoying, they make the logs unuseable as they persist at a rate of 4 per second forever.

grozzie2 replied:
I will add to that, since they are continually coming out of the serial console as well, the serial console is almost unuseable too.

robimarko replied:
Yeah, I get the issue but just adding fake PHY-s is not the solution.
I will eventually look into the issue, maybe over the weekend to at least add a feature to disable the port if not used.

musashino205 commented on package/firmware/ipq-wifi/Makefile:
```diff
@@ -40,13 +40,19 @@ ALLWIFIBOARDS:= \
 
 ALLWIFIPACKAGES:=$(foreach BOARD,$(ALLWIFIBOARDS),ipq-wifi-$(BOARD))
 
+define Build/Prepare
+	mkdir -p $(PKG_BUILD_DIR)
+	cp board/* $(PKG_BUILD_DIR)
+endef
+
```
Why did you add this?

legale replied:
Due to the lack of the correct BDF file for the device in the corresponding repo (qca-wireless).

I posted the BDF file to the ath11k mailing list. Until the file for the device is added to the related repo, it is located here. It is copied in addition to the others, which are downloaded from the qca-wireless repo.

musashino205 replied:
I'm talking about ```Build/Prepare``` section...

legale replied:
8-) 


This recipe copies the BDF file for the Yuncore AX880 device (https://github.com/openwrt/openwrt/blob/9297dfcc4d6263612f6b399be7f043ea4daa88cb/package/firmware/ipq-wifi/board/board-yuncore_ax880.ipq8074) to the qca-wireless package directory, from where this file is later copied to the compiled firmware with this recipe `ipq-wifi-install-ath11-one-to` (https://github.com/openwrt/openwrt/blob/9297dfcc4d6263612f6b399be7f043ea4daa88cb/package/firmware/ipq-wifi/Makefile#L69).

robimarko replied:
Yeah, this should be dropped as we are intentionally shipping the BDF-s in a separate repo.
I think I saw you sent the BDF as a patch instead of a PR on GH?

legale replied:
If i remove this recipe before related BDF will be accepted automatic check will fail. What to do?

robimarko commented on target/linux/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata:
```diff
@@ -21,6 +21,9 @@ case "$FIRMWARE" in
 	zyxel,nbg7815)
 		caldata_extract "0:art" 0x1000 0x20000
 		;;
+    yuncore,ax880)
```
Broken indent.

legale replied:
fixed.

robimarko commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,390 @@
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
```
Are the partitions defined in SMEM incorrect?

legale replied:
These are correct for sure.

kirdesde replied:
Just to make, can you please post the output of smem in u-boot to see if the SMEM table is correct or broken?

legale replied:
> Just to make, can you please post the output of smem in u-boot to see if the SMEM table is correct or broken?

```
IPQ807x# smem
flash_type:             0x6
flash_index:            0x0
flash_chip_select:      0x0
flash_block_size:       0x10000
flash_density:          0x1000000
partition table offset  0x0
No.: Name             Attributes            Start             Size
  0: 0:SBL1           0x0000ffff              0x0          0x50000
  1: 0:MIBIB          0x001040ff          0x50000          0x10000
  2: 0:BOOTCONFIG     0x001040ff          0x60000          0x20000
  3: 0:BOOTCONFIG1    0x001040ff          0x80000          0x20000
  4: 0:QSEE           0x0000ffff         0x220000         0x180000
  5: 0:QSEE_1         0x0000ffff          0xa0000         0x180000
  6: 0:DEVCFG         0x0000ffff         0x3b0000          0x10000
  7: 0:DEVCFG_1       0x0000ffff         0x3a0000          0x10000
  8: 0:APDP           0x0000ffff         0x3c0000          0x10000
  9: 0:APDP_1         0x0000ffff         0x3d0000          0x10000
 10: 0:RPM            0x0000ffff         0x420000          0x40000
 11: 0:RPM_1          0x0000ffff         0x3e0000          0x40000
 12: 0:CDT            0x0000ffff         0x470000          0x10000
 13: 0:CDT_1          0x0000ffff         0x460000          0x10000
 14: 0:APPSBLENV      0x0000ffff         0x480000          0x10000
 15: 0:APPSBL         0x0000ffff         0x530000          0xa0000
 16: 0:APPSBL_1       0x0000ffff         0x490000          0xa0000
 17: 0:ART            0x0000ffff         0x5d0000          0x40000
 18: 0:ETHPHYFW       0x0000ffff         0x610000          0x80000
 19: rootfs           0x0100ffff        0x3c00000        0x3400000
 20: 0:WIFIFW         0x0100ffff        0x7000000         0x800000
 21: rootfs_1         0x0100ffff              0x0        0x3400000
 22: 0:WIFIFW_1       0x0100ffff        0x3400000         0x800000
IPQ807x#
```

legale replied:
If i remove these partiions in dts there is no chance to work with them from initramfs, so i think it is better to leave them.

kirdesde replied:
Okay I see, using smem would lead to overlapping partitions.
But any particular reason why you are missing lowercase and uppercase for the partition labels?

In my opinion you should use lowercase for all labels.

robimarko replied:
I agree, please lowercase them so userspace scripts dont have to be changed just because of upper case.

I get the issue, it will try to use the same table for both NOR and NAND as there is no flash type parsing.

robimarko commented on target/linux/ipq807x/image/generic.mk:
```diff
@@ -173,3 +173,20 @@ define Device/zyxel_nbg7815
+	DEVICE_PACKAGES := ipq-wifi-yuncore_ax880
+	IMAGES := sysupgrade.bin factory.bin factory.ubi	
+	IMAGE/factory.bin := append-ubi | qsdk-ipq-factory-nand
+	IMAGE/factory.ubi := append-ubi
```
No need to override the recipe defined in `Device/UbiFit` as its the same

legale replied:
remove these lines?

```
	IMAGE/factory.bin := append-ubi | qsdk-ipq-factory-nand
	IMAGE/factory.ubi := append-ubi
```

robimarko commented on target/linux/ipq807x/image/generic.mk:
```diff
@@ -173,3 +173,20 @@ define Device/zyxel_nbg7815
+	DEVICE_DTS_CONFIG := config@hk09
+	SOC := ipq8072
+	DEVICE_PACKAGES := ipq-wifi-yuncore_ax880
+	IMAGES := sysupgrade.bin factory.bin factory.ubi	
```
`IMAGES += factory.bin` doesn't work?

legale replied:
factory.bin is not enough. U-boot allow to flash factory.bin but it is a lot more complicated. 

To flash ubi file there only 2 commands needed:
```
tftpboot server_ip:file.ubi
flash rootfs
```
So i prefer factory.ubi

robimarko replied:
You are not getting my point, `Device/UbiFit` already defines `IMAGES := factory.ubi sysupgrade.bin`.
So my question was whether simply extending the variable via `IMAGES += factory.bin` works instead of redefining it?

robimarko replied:
Also, where is the recipe for `factory.bin`, cause I dont see one at all?

legale replied:
> You are not getting my point, `Device/UbiFit` already defines `IMAGES := factory.ubi sysupgrade.bin`. So my question was whether simply extending the variable via `IMAGES += factory.bin` works instead of redefining it?

Now i got it. Append factory.bin

robimarko commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,390 @@
+
+&wifi {
+	status = "okay";
+	qcom,board_id = <0xff>;
```
This is unused, and default anyway.

legale replied:
remove?

kirdesde replied:
You haven't removed the qcom,board_id property.

legale replied:
> You haven't removed the qcom,board_id property.
Missed it. Thank you.
Removed.

robimarko commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,390 @@
+
+		wlan2g {
+			label = "wlan2g";
+			gpio = <&tlmm 2 GPIO_ACTIVE_HIGH>;
```
No trigger?

Please also specify the color functionality, label should include the color as well

legale replied:
fixed.

robimarko commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,390 @@
+};
+
+&blsp1_uart5 {
+	status = "okay";
```
Is this being used?

legale replied:
yes.

legale commented:
@musashino205 @robimarko @hauke @kirdesde

Everything is done. Please review.

kirdesde commented:
> @musashino205 @robimarko @hauke @kirdesde @Ansuel @aiamadeus
> 
> Everything is done. Please review.

There are still a few things to be done, I'v left a few comments in the ( as resolved marked) discussions

legale commented:
@robimarko @musashino205 @Ansuel @kirde

Everything is done, please review.

> > @musashino205 @robimarko @hauke @kirdesde @Ansuel @aiamadeus
> > Everything is done. Please review.
> 
> There are still a few things to be done, I'v left a few comments in the ( as resolved marked) discussions

Ok. Thanks.

kirdesde commented:
Please use lowercase for all partition labels (you need to adjust the 11-ath11k-caldata then, you can just add your device to the existing device list under the zyxel,nbg7815 entry)

And please add some install instructions to the device description.

Anything else LGTM

legale commented:
No. I mean these:


пт, 3 нояб. 2023 г., 07:37 kirdesde ***@***.***>:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts
> <https://github.com/openwrt/openwrt/pull/13853#discussion_r1381154196>:
>
> > +    ***@***.*** {
> +        reg = <0>;
> +    };
> +    ***@***.*** {
> +        reg = <1>;
> +    };
> +    ***@***.*** {
> +        reg = <2>;
> +    };
> +    ***@***.*** {
> +        reg = <3>;
> +    };
>
> You mean these log entries?
>
> phy_id = 0xffffffff phytype doesn't match
>
> Those can be ignored, they should only occur during ssdk init.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/13853#discussion_r1381154196>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHY4JF5X5C3NUFAPTPECC6TYCRYHXAVCNFSM6AAAAAA62G7URKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMJRHAZDAOJUGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

robimarko commented:
> Please use lowercase for all partition labels (you need to adjust the 11-ath11k-caldata then, you can just add your device to the existing device list under the zyxel,nbg7815 entry)
> 
> And please add some install instructions to the device description.
> 
> Anything else LGTM

I second this, will give it another look after this change.

robimarko commented on target/linux/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata:
```diff
@@ -21,23 +21,26 @@ case "$FIRMWARE" in
+		;;
 	prpl,haze)
-                caldata_extract_mmc "0:ART" 0x1000 0x20000
+		caldata_extract_mmc "0:ART" 0x1000 0x20000
```
I am sure this is unintended, remove the change.

legale replied:
I don't get it.

Case blocks content is intended now:
https://github.com/openwrt/openwrt/blob/ae500e62e2938e112ae1fc6aa7389e8c7b784b13/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata#L33

I have replaced all spaces with tabs (some of them are present in this file at the moment https://github.com/openwrt/openwrt/blob/ae500e62e2938e112ae1fc6aa7389e8c7b784b13/target/linux/qualcommax/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata#L37C15-L37C15).

robimarko commented on target/linux/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata:
```diff
@@ -21,23 +21,26 @@ case "$FIRMWARE" in
 	case "$board" in
 	prpl,haze)
-                caldata_extract_mmc "0:ART" 0x26800 0x20000
+		caldata_extract_mmc "0:ART" 0x26800 0x20000
```
Same here

legale replied:
sorry for this. I thought i've replaced 4 spaces with tabs everywhere.

legale replied:
Done.

robimarko commented on target/linux/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata:
```diff
@@ -21,23 +21,26 @@ case "$FIRMWARE" in
-	;;
+	*)
+		exit 1
+		;;
```
Same for this block as well

legale replied:
Done.

robimarko commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,367 @@
+
+		wlan2g {
+			label = "green:wlan2g";
+			linux,default-trigger = "phy0radio";
```
Have you tested this?
`phy0radio` usually doesn't work well, `phy0tpt` usually works better.

Same for radio1

legale replied:
ok. Haven't tested it yet. Will replace phy0radio with phy0tpt

legale replied:
Done.

robimarko commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,367 @@
+		wlan2g {
+			label = "green:wlan2g";
+			linux,default-trigger = "phy0radio";
+			gpios = <&tlmm 2 GPIO_ACTIVE_HIGH>;
```
Still no color property being defined, we want to eventually get rid of labels so please include the color.

legale replied:
ok.

legale replied:
Done.

legale commented:
> Please use lowercase for all partition labels (you need to adjust the 11-ath11k-caldata then, you can just add your device to the existing device list under the zyxel,nbg7815 entry)
> 
> And please add some install instructions to the device description.
> 
> Anything else LGTM

i got it.

legale commented:
> Please use lowercase for all partition labels (you need to adjust the 11-ath11k-caldata then, you can just add your device to the existing device list under the zyxel,nbg7815 entry)
> 
> And please add some install instructions to the device description.
> 
> Anything else LGTM

@kirdesde 

partitions names now in lowercase. Related changes in `target/linux/ipq807x/base-files/etc/hotplug.d/firmware/11-ath11k-caldata` are made.

legale commented:
@kirdesde @robimarko @musashino205
Please review again.

kirdesde commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,370 @@
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "0:SBL1";
```
lowercase please

legale replied:
done.

kirdesde commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,370 @@
+			};
+
+			partition@3400000 {
+				label = "0:WIFIFW";
```
lowercase please

legale replied:
done.

kirdesde commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,370 @@
+			};
+
+			partition@7000000 {
+				label = "0:WIFIFW_1";
```
lowercase please

legale replied:
done.

kirdesde commented:
> @kirdesde @robimarko @musashino205 Please review again.

There are 3 uppercase partition labels left.


Are you sure that the described initramfs method installation actually works?

Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)

legale commented:
> > @kirdesde @robimarko @musashino205 Please review again.
> 
> There are 3 uppercase partition labels left.
> 
> Are you sure that the described initramfs method installation actually works?
> 
> Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)

I will try.

> > @kirdesde @robimarko @musashino205 Please review again.
> 
> There are 3 uppercase partition labels left.
> 
> Are you sure that the described initramfs method installation actually works?
> 
> Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)

Your truth. Thanks.
Updated initramfs method and added another one (stock fw method). Added more useful info.

kirdesde commented:
> @kirdesde @robimarko @musashino205 Please review again.

There are 3 uppercase partition labels left.


Are you sure that the described initramfs method installation actually works?

Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)

> > > @kirdesde @robimarko @musashino205 Please review again.
> > 
> > 
> > There are 3 uppercase partition labels left.
> > Are you sure that the described initramfs method installation actually works?
> > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> 
> I will try.
> 
> > > @kirdesde @robimarko @musashino205 Please review again.
> > 
> > 
> > Are you sure that the described initramfs method installation actually works?
> > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> 
> Your truth. Thanks. Updated initramfs method and added another one (stock fw method). Added more useful info.
> 
> P.S. These methods are tested couple of minutes ago.

Regarding "3. initramfs method"

Sorry, but this is important:

Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)

Use ubiformat instead (that just formats the ubi on top of the NAND)

like that:

```
cat /proc/mtd 
```
and find the corresponding rootfs and rootfs_1 entries.

then:

```
ubiformat /dev/mtdx -y -f factory.ubi (replace x with the mtd num for rootfs/rootfs_1)
```

Anything else LGTM

legale commented:
> > @kirdesde @robimarko @musashino205 Please review again.
> 
> There are 3 uppercase partition labels left.
> 
> Are you sure that the described initramfs method installation actually works?
> 
> Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> 
> > > > @kirdesde @robimarko @musashino205 Please review again.
> > > 
> > > 
> > > There are 3 uppercase partition labels left.
> > > Are you sure that the described initramfs method installation actually works?
> > > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> > 
> > 
> > I will try.
> > > > @kirdesde @robimarko @musashino205 Please review again.
> > > 
> > > 
> > > Are you sure that the described initramfs method installation actually works?
> > > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> > 
> > 
> > Your truth. Thanks. Updated initramfs method and added another one (stock fw method). Added more useful info.
> > P.S. These methods are tested couple of minutes ago.
> 
> Regarding "3. initramfs method"
> 
> Sorry, but this is important:
> 
> Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)
> 
> Use ubiformat instead (that just formats the ubi on top of the NAND)
> 
> like that:
> 
> ```
> cat /proc/mtd 
> ```
> 
> and find the corresponding rootfs and rootfs_1 entries.
> 
> then:
> 
> ```
> ubiformat /dev/mtdx -y -f factory.ubi (replace x with the mtd num for rootfs/rootfs_1)
> ```
> 
> Anything else LGTM

I got it. 
Tell me please more about mtd erase/write for nand.

kirdesde commented:
> @kirdesde @robimarko @musashino205 Please review again.

There are 3 uppercase partition labels left.


Are you sure that the described initramfs method installation actually works?

Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)

> > > @kirdesde @robimarko @musashino205 Please review again.
> > 
> > 
> > There are 3 uppercase partition labels left.
> > Are you sure that the described initramfs method installation actually works?
> > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> 
> I will try.
> 
> > > @kirdesde @robimarko @musashino205 Please review again.
> > 
> > 
> > Are you sure that the described initramfs method installation actually works?
> > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> 
> Your truth. Thanks. Updated initramfs method and added another one (stock fw method). Added more useful info.
> 
> P.S. These methods are tested couple of minutes ago.

Regarding "3. initramfs method"

Sorry, but this is important:

Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)

Use ubiformat instead (that just formats the ubi on top of the NAND)

like that:

```
cat /proc/mtd 
```
and find the corresponding rootfs and rootfs_1 entries.

then:

```
ubiformat /dev/mtdx -y -f factory.ubi (replace x with the mtd num for rootfs/rootfs_1)
```

Anything else LGTM

> > > @kirdesde @robimarko @musashino205 Please review again.
> > 
> > 
> > There are 3 uppercase partition labels left.
> > Are you sure that the described initramfs method installation actually works?
> > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> > > > > @kirdesde @robimarko @musashino205 Please review again.
> > > > 
> > > > 
> > > > There are 3 uppercase partition labels left.
> > > > Are you sure that the described initramfs method installation actually works?
> > > > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> > > 
> > > 
> > > I will try.
> > > > > @kirdesde @robimarko @musashino205 Please review again.
> > > > 
> > > > 
> > > > Are you sure that the described initramfs method installation actually works?
> > > > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> > > 
> > > 
> > > Your truth. Thanks. Updated initramfs method and added another one (stock fw method). Added more useful info.
> > > P.S. These methods are tested couple of minutes ago.
> > 
> > 
> > Regarding "3. initramfs method"
> > Sorry, but this is important:
> > Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)
> > Use ubiformat instead (that just formats the ubi on top of the NAND)
> > like that:
> > ```
> > cat /proc/mtd 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and find the corresponding rootfs and rootfs_1 entries.
> > then:
> > ```
> > ubiformat /dev/mtdx -y -f factory.ubi (replace x with the mtd num for rootfs/rootfs_1)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Anything else LGTM
> 
> I got it. Tell me please more about mtd erase/write for nand.

mtd doesn't deal with bad blocks on NAND. 

see https://openwrt.org/docs/techref/flash#nand-specific_tools_for_reading_and_writing_to_raw_nand

legale commented:
> > @kirdesde @robimarko @musashino205 Please review again.
> 
> ```
> ubiformat /dev/mtdx -y -f factory.ubi (replace x with the mtd num for rootfs/rootfs_1)
> ```
> 
> Anything else LGTM

I have replaced `mtd erase/write` with `ubiformat`

legale commented:
> Sorry, but this is important:
> 
> Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)
> 
> Use ubiformat instead (that just formats the ubi on top of the NAND)

If you are talking about nand ecc i'm sure that this device uses hardware ecc, so there is no difference what exactly to use. Correct me if i'm wrong.

kirdesde commented:
> @kirdesde @robimarko @musashino205 Please review again.

There are 3 uppercase partition labels left.


Are you sure that the described initramfs method installation actually works?

Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)

> > > @kirdesde @robimarko @musashino205 Please review again.
> > 
> > 
> > There are 3 uppercase partition labels left.
> > Are you sure that the described initramfs method installation actually works?
> > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> 
> I will try.
> 
> > > @kirdesde @robimarko @musashino205 Please review again.
> > 
> > 
> > Are you sure that the described initramfs method installation actually works?
> > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> 
> Your truth. Thanks. Updated initramfs method and added another one (stock fw method). Added more useful info.
> 
> P.S. These methods are tested couple of minutes ago.

Regarding "3. initramfs method"

Sorry, but this is important:

Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)

Use ubiformat instead (that just formats the ubi on top of the NAND)

like that:

```
cat /proc/mtd 
```
and find the corresponding rootfs and rootfs_1 entries.

then:

```
ubiformat /dev/mtdx -y -f factory.ubi (replace x with the mtd num for rootfs/rootfs_1)
```

Anything else LGTM

> > > @kirdesde @robimarko @musashino205 Please review again.
> > 
> > 
> > There are 3 uppercase partition labels left.
> > Are you sure that the described initramfs method installation actually works?
> > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> > > > > @kirdesde @robimarko @musashino205 Please review again.
> > > > 
> > > > 
> > > > There are 3 uppercase partition labels left.
> > > > Are you sure that the described initramfs method installation actually works?
> > > > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> > > 
> > > 
> > > I will try.
> > > > > @kirdesde @robimarko @musashino205 Please review again.
> > > > 
> > > > 
> > > > Are you sure that the described initramfs method installation actually works?
> > > > Usually it's not enough to sysupgrade from the running initramfs, instead an ubiformat is necessary due to qsdk based firmware uses "ubi_rootfs" as ubi volume name for volume ID 1 in contrast to OpenWrt ("rootfs" ubi volume name)
> > > 
> > > 
> > > Your truth. Thanks. Updated initramfs method and added another one (stock fw method). Added more useful info.
> > > P.S. These methods are tested couple of minutes ago.
> > 
> > 
> > Regarding "3. initramfs method"
> > Sorry, but this is important:
> > Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)
> > Use ubiformat instead (that just formats the ubi on top of the NAND)
> > like that:
> > ```
> > cat /proc/mtd 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and find the corresponding rootfs and rootfs_1 entries.
> > then:
> > ```
> > ubiformat /dev/mtdx -y -f factory.ubi (replace x with the mtd num for rootfs/rootfs_1)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Anything else LGTM
> 
> I got it. Tell me please more about mtd erase/write for nand.

mtd doesn't deal with bad blocks on NAND. 

see https://openwrt.org/docs/techref/flash#nand-specific_tools_for_reading_and_writing_to_raw_nand

> > Sorry, but this is important:
> > Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)
> > Use ubiformat instead (that just formats the ubi on top of the NAND)
> 
> If you are talking about nand ecc i'm sure that this device uses hardware ecc, so there is no difference what exactly to use. Correct me if i'm wrong.

Bad block management and ECC are 2 different things.

ECC is used to correct temporary errors during read/write (bit-flipping).

Bad block management: NAND flash always contains bad blocks, they are marked in a table and they must read before every erase operation to avoid erasing them 'cause that can damaged even good blocks.

legale commented:
> mtd doesn't deal with bad blocks on NAND.
> 
> see https://openwrt.org/docs/techref/flash#nand-specific_tools_for_reading_and_writing_to_raw_nand
> 
> > > Sorry, but this is important:
> > > Do not use mtd erase/mtd write on NAND (this is only safe for NOR flash)
> > > Use ubiformat instead (that just formats the ubi on top of the NAND)
> > 
> > 
> > If you are talking about nand ecc i'm sure that this device uses hardware ecc, so there is no difference what exactly to use. Correct me if i'm wrong.
> 
> Bad block management and ECC are 2 different things.
> 
> ECC is used to correct temporary errors during read/write (bit-flipping).
> 
> Bad block management: NAND flash always contains bad blocks, they are marked in a table and they must read before every erase operation to avoid erasing them 'cause that can damaged even good blocks.

Thanks.

kirdesde commented:
Reviewed-by: Dirk Buchwalder <buchwalder@posteo.de>

legale commented:
@musashino205 @robimarko @Ansuel @hauke @rmilecki @skramstad @ynezz 

Could you review updated PR please.

robimarko commented:
I dont see anything obviously wrong so:
`Reviewed-by: Robert Marko <robimarko@gmail.com>`

legale commented:
just git rebase openwrt-23.05 

Nothing changed.

robimarko commented:
Oh, wait, I just looked and you are targeting OpenWrt 23.05.
Please target master/main instead.

grozzie2 reviewed (approved):
I have two of these boards here, the indoor variant.  Building with this does indeed bring the board up, wireless and ethernet working, works well on my 2.5gb switch.

As I commented earlier, the endless log spam from the qsdk code is annoying as it continues to try find hardware that is into present on the board, but I believe the only cure for that is to fix the sdk and not this set of board specific changes.

Just noticed the comment about 23.05 from robimarko.  When I built, I cloned legale repository using his ax880 branch.  Struggling here to figure out how to 'undo' the reviewed tag

robimarko reviewed (changes requested):
Please rebase everything on top of master.

rmilecki commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,370 @@
+				read-only;
+			};
+
+			rootfs: partition@3c00000 {
```
Is that label `rootfs` used anywhere?

rmilecki commented on target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts:
```diff
@@ -0,0 +1,370 @@
+// SPDX-License-Identifier: MIT
```
Dual licensing is safer as compatibility isn't based on current licenses interpretation. Do you think you could release it as `GPL-2.0-or-later OR MIT`?

legale replied:
no problem.

legale commented:
Ok.

чт, 9 нояб. 2023 г., 13:29 Robert Marko ***@***.***>:

> ***@***.**** requested changes on this pull request.
>
> Please rebase everything on top of master.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/13853#pullrequestreview-1722219074>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHY4JFYRK5TGHJSTQL6H6QTYDSWB7AVCNFSM6AAAAAA62G7URKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMRSGIYTSMBXGQ>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

legale commented:
this pr replaced with: https://github.com/openwrt/openwrt/pull/13918

Ansuel commented:
@legale don't want to be rude but why the pr replacement? we lose all history... you can change the base branch on the top near the pr title. 

Please close the replacement pr and lets get back here! (we are starting to flag pr as merged so it makes sense as they gets linked to the merged commit)

Ansuel commented:
@legale if you need help feel free to ask ahahah (i know it's hell to handle this)

legale commented:
> @legale if you need help feel free to ask ahahah (i know it's hell to handle this)

Help please. 

How to handle this:

![image](https://github.com/openwrt/openwrt/assets/32621719/778c7299-033b-4cf2-88a9-ca321dbc4115)

602 commits appears after i have changed branch from openwrt-23.05 to master

Ansuel commented:
got where you have your owrt-master-ax880 or checkout to this branch

`git push origin HEAD:owrt-23.05-ax880 -f`

Ansuel commented:
@robimarko looks all ok to me. If you are ok I would proceed with merging this.

robimarko commented:
LGTM,
`Reviewed-by: Robert Marko <robimarko@gmail.com>`

Ansuel commented:
@legale can you add the review tag so i can merge? :D

legale commented:
> @legale can you add the review tag so i can merge? :D

Where is this button?

Ansuel commented:
@legale I will take care of it :D 

(just for reference
1. you should drop the last commit
2. to add stuff to an already pushed commit git commit --amend
3. add the tag robimarko reported
4. force push again
)

legale commented:
> @legale I will take care of it :D
> 
> (just for reference
> 
> 1. you should drop the last commit
got it
> 2. to add stuff to an already pushed commit git commit --amend
got it
> 3. add the tag robimarko reported
didn't get it. 
> 4. force push again
got it
>    )

Neustradamus commented:
@legale: Good job!

zhouruixi commented:
same issue with Xiaomi AX3600 https://github.com/svpcom/wfb-ng/issues/365
